### DISCUSSION TITLE
Automatically reconnect sessions.

### DISCUSSION THREAD

**Discussion by rjbmc** [community/community#88657] [Programming Help]
### Body

I’m using Hibernate + C3P0 reading an Oracle database in a Tomcat 9 environment. 
Is it possible to automatically reconnect sessions when they have been closed due to inactive timeout?. 
What parameters should be set to persistence.xml?
I've read about this point but only in Spring framework, whitout Spring I have'nt found nothing.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by AidenJamesYt** [comment]
n a Hibernate and C3P0 configuration without Spring, you can set certain properties to handle the automatic reconnection of sessions in case they have been closed due to inactivity. To achieve this, you'll need to configure the hibernate.c3p0.* properties in your persistence.xml.
